addon/components/bs-modal.js

```diff
@@ -137,6 +117,25 @@ export default class Modal extends Component {
   @tracked
   paddingRight;
 
+  /**
+   * Visibility of the modal. Toggle to show/hide with CSS transitions.
+   *
+   * When the modal is closed by user interaction this property will not update by using two-way bindings in order
+   * to follow DDAU best practices. If you want to react to such changes, subscribe to the `onHide` action
+   *
+   * @property open
+   * @type boolean
+   * @default true
+   * @public
+   */
+
+  /**
+   * @property open
+   * @private
+   */
+  @arg
+  open = true;
```

Comment:
Any reason you renamed the variable from `isOpen` to `open`? The `is` prefix indicates that it's a boolean. I fear `open` can be confused with a method to open the modal. I would prefer to keep `isOpen` to be honest. Would it be okay for you to change back?

Reply:
I haven't reviewed this PR yet, but I am a bit confused at this point: `@open` is the public argument, which we cannot rename. The JS docs above are wrong somehow, as there is `@property open` twice!

Reply:
I've tried to rename `isOpen` to `open` to remove the dependency on the @localCopy and to clarify that it is an argument to the `BsModal` component. I must admit that I do not understand why there is both a `isOpen` and a `_isOpen`. The JSDoc seems indeed to be a bit odd, as `isOpen` had two JSDocs above it as if something between the docs was missing. 

While I do agree that `isOpen` would be a better name for a boolean, it does not reflect the actual args of the component well, causing confusion. I've tried a few ways of getting it to be a proper alias of open, but none of them were succesful.

Reply:
Oh. I missed that this has `@arg` decorator pointing to the argument. Sorry. Should have taken more time to review. Can not could how often I already got confused by `@open` argument and the two internal properties `isOpen` and `_isOpen`.

To be honest I thought this PR decouples `@open` from the internal state and manages the state updates through `{{did-update}}` modifier in the template only. I think I need to reserve some time to fully understand state flow in this component.

Reply:
I'm not fully clear about the lifecycle of this component. I mostly noticed that the `_isOpen` state did not get put back to false, causing the modal to never reopen. It does seem to fix the odd behaviour people have been experiencing without breaking the existing test conditions.